apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -0,0 +1,522 @@
+use crate::plugins::telemetry::config::Trace;
+use crate::plugins::telemetry::tracing::apollo;
+use crate::subscriber::replace_layer;
+use apollo_router_core::{register_plugin, Plugin};
+use apollo_spaceport::server::ReportSpaceport;
+use futures::FutureExt;
+use opentelemetry::propagation::TextMapPropagator;
+use opentelemetry::sdk::propagation::{
+    BaggagePropagator, TextMapCompositePropagator, TraceContextPropagator,
+};
+use opentelemetry::sdk::trace::Builder;
+use opentelemetry::sdk::Resource;
+use opentelemetry::trace::TracerProvider;
+use opentelemetry::{global, Array, KeyValue, Value};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+use std::error::Error;
+use std::fmt;
+use tower::BoxError;
+use url::Url;
+
+pub mod metrics;
+pub mod tracing;
+
+pub struct Telemetry {
+    config: config::Conf,
+    tracer_provider: Option<opentelemetry::sdk::trace::TracerProvider>,
+    spaceport_shutdown: Option<futures::channel::oneshot::Sender<()>>,
+}
+pub mod config {
```

Comment:
Why not putting these configs in a `config.rs` ?

Reply:
Fixed in [e033fdf](https://github.com/apollographql/router/pull/782/commits/e033fdf27be319c6bc77518f2509d6e5fa5c8dd8)